spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -2378,14 +2416,38 @@ public void doSend(Channel channel, String exchangeArg, String routingKeyArg, Me
 			logger.debug("Publishing message [" + messageToUse
 					+ "] on exchange [" + exch + "], routingKey = [" + rKey + "]");
 		}
-		sendToRabbit(channel, exch, rKey, mandatory, messageToUse);
+		observeTheSend(channel, message, mandatory, exch, rKey, messageToUse);
 		// Check if commit needed
 		if (isChannelLocallyTransacted(channel)) {
 			// Transacted channel created by this template -> commit.
 			RabbitUtils.commitIfNecessary(channel);
 		}
 	}
 
+	protected void observeTheSend(Channel channel, Message message, boolean mandatory, String exch, String rKey,
+			Message messageToUse) {
```

Comment:
Why do we need two `Message` instances over here?
Why just one `messageToUse` from the previous modification is not enough?

Reply:
Refactor error.